invokeai/app/invocations/params.py

```diff
@@ -2,17 +2,131 @@
         return IntOutput(a=self.a)
+
+
+class FloatOutput(BaseInvocationOutput):
+    """A float output"""
+    #fmt: off
+    type: Literal["float_output"] = "float_output"
+    a: float = Field(default=None, description="The output float value")
+    #fmt: on
+
+
+class ParamFloatInvocation(BaseInvocation):
+    """A float parameter"""
+    #fmt: off
+    type: Literal["param_float"] = "param_float"
+    a: float = Field(default=0, description="A float value")
+    #fmt: on
+
+    class Config(InvocationConfig):
+        schema_extra = {
+            "ui": {
+                "tags": ["parameters", "float"],
+                "title": "Float Value"
+            }
+        }
+
+    def invoke(self, context: InvocationContext) -> FloatOutput:
+        return FloatOutput(a=self.a)
+
+
+class StringOutput(BaseInvocationOutput):
+    """A string output"""
+    #fmt: off
+    type: Literal["string_output"] = "string_output"
+    a: str = Field(default=None, description="The output string value")
+    #fmt: on
+
+
+class ParamStringInvocation(BaseInvocation):
+    """A string parameter"""
+    #fmt: off
+    type: Literal["param_string"] = "param_string"
+    a: str = Field(default='', description="A string value")
+    #fmt: on
+
+    class Config(InvocationConfig):
+        schema_extra = {
+            "ui": {
+                "tags": ["parameters", "string"],
+                "title": "String Value"
+            }
+        }
+
+    def invoke(self, context: InvocationContext) -> StringOutput:
+        return StringOutput(a=self.a)
+
+
+class BooleanOutput(BaseInvocationOutput):
+    """A boolean output"""
+    #fmt: off
+    type: Literal["boolean_output"] = "boolean_output"
+    a: bool = Field(default=None, description="The output boolean value")
+    #fmt: on
+
+
+class ParamBooleanInvocation(BaseInvocation):
+    """A boolean parameter"""
+    #fmt: off
+    type: Literal["param_boolean"] = "param_boolean"
+    a: bool = Field(default=False, description="A boolean value")
+    #fmt: on
+
+    class Config(InvocationConfig):
+        schema_extra = {
+            "ui": {
+                "tags": ["parameters", "boolean"],
+                "title": "Boolean Value"
+            }
+        }
+
+    def invoke(self, context: InvocationContext) -> BooleanOutput:
+        return BooleanOutput(a=self.a)
+
+class ParamMergeStringInvocation(BaseInvocation):
+    """Merge two string nodes"""
+    type: Literal["merge_param_string"] = "merge_param_string"
+    a: str = Field(default='', description='First string to merge')
+    b: str = Field(default='', description='Second string to merge')
+
+    class Config(InvocationConfig):
+        schema_extra = {
+            "ui": {
+                "tags": ["parameters", "merge", "string"],
+                "title": "Merge String"
+            }
+        }
+
+    def invoke(self, context: InvocationContext) -> StringOutput:
+        return StringOutput(a=f'{self.a} {self.b}')
```

Comment:
I don't think this should be a node - we should design a more generalized way to concatenate strings, ideally using multiple iterate/collect nodes (or similar), which let you do distributed concatenation over multiple arrays. 

The simplest use case of such a node would be concatenating two strings, superseding this node, but it could also do much more.

Reply:
Okie removing it.